emma-flink/src/main/scala/org/emmalanguage/compiler/flink/FlinkSpecializeSupport.scala

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright © 2014 TU Berlin (emma@dima.tu-berlin.de)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.emmalanguage
+package compiler.flink
+
+import compiler.FlinkCompiler
+import compiler.ir.DSCFAnnotations._
+
+import shapeless._
+
+import scala.annotation.tailrec
+import scala.collection.breakOut
+
+private[compiler] trait FlinkSpecializeSupport {
+  self: FlinkCompiler =>
+
+  import API.DataBag
+  import Core.{Lang => core}
+  import FlinkAPI.Ntv
+  import UniverseImplicits._
+
+  object FlinkSpecializeSupport {
+
+    /** Introduces Flink's native iterations whenever possible. */
```

Comment:
Could you please extend this comment with the conditions of when is it possible?

E.g.:
- no nested loops or if-statements inside
- only one bag is passed between subsequent iterations
- and this bag is the only thing that comes out from the loop to the suffix
- no side-effecting operations in the loop (e.g. DataBag IO)
- loop body can be completely translated into a single Flink dataflow:
  - there are no non-DataBag operations (except for the loop counter variable?)
  - no `collect` is necessary
- fixed number of iterations (this is just a limitation of the current implementation, right?)

Reply:
Done. The code still has come rough edges (I'm in *deadline* mode) but should do the job for most scenarios.